### PR TITLE
Add "hidden" property attribute

### PR DESCRIPTION
**Pull request by gramian**
## What does this PR do?

This new boolean property attribute called `hidden` excludes properties from a projection containing a `*` (star).

## Motivation

[ZetaSQL](https://github.com/google/zetasql/blob/master/docs/data-definition-language.md?plain=1#L204)

## Related issues

https://github.com/ArcadeData/arcadedb/discussions/1439

## Additional Notes

This is the interesting change set: https://github.com/ArcadeData/arcadedb/pull/1503/files#diff-1ebfff8995bc54eab127a05a7f09d799b215972c6aa872635f37d86e824fb0b3

* Basically now the any projection with a `*` (`isAll`) gets parsed property by property not by copy any more.
* `SELECT * FROM x` works, but `SELECT FROM x` does not.
    * @lvca I cannot figure out why this is not working. 

## Checklist

- [ ] I have run the build using `mvn clean package` command
- [ ] My unit tests cover both failure and success scenarios

### CODE REVIEW

**Comment by robfrank** [comment]
@gramian  I added the missing part, essentially I updated the sql grammar and added some tests. 

So, how it works?

If a property is configure to be hidden, when using the 

```sql
SELECT * FROM TypeWithHiddenPropery
```
than the hidden property isn't present in the result set.

But, when not using the `*`:
```sql
SELECT  FROM TypeWithHiddenPropery
```
than the hidden property IS present in the result set.

After a fast call with @lvca, we think this is the right way to implement the feature. When a the query is written without a projection, the `*` in this case, it means the the query should fetch the whole content, hidded properties included. 

WDYT?

**Comment by gramian** [comment]
Hmm, OK.

I always thought of `SELECT FROM` as short form of `SELECT * FROM`. Are there other differences between these two forms?

**Comment by robfrank** [comment]
They are working differently in the parser. The `*` generates a   projection that is then interpreted as "all" , while the "blank" doesn't generate any projection.
On performace side, I think it depends more on the whole query structure rather than a  single part of the statement.

**Comment by gramian** [comment]
It definitely is.

I am thinking about documentation for users, so when to use which and why. So given a large document, meaning many properties, `SELECT FROM` would be slightly faster as just the entire document is returned, right? And this ties now in with `hidden`; because the whole doc is returned the hidden properties are included, whereas using `SELECT * FROM` computes a projection in which `hidden` is applied. What else needs to be stated in this regard?

**Comment by robfrank** [comment]
exactly: you get to the point.

**Review by lvca — APPROVED** [review]
